src/ast/desugar.rs

```diff
@@ -339,6 +358,7 @@ pub(crate) fn desugar_command(
             };
 
             desugar.desugar_program(
+                // TODO: all spans should be that of the original query
                 desugar.parse_program(None, &desugaring).unwrap(),
```

Comment:
We could panic when this unwrap fails, and point to the span for the query extract. It would be an okay way to fix the error message here

Reply:
This would not work in a lot of cases since the failure usually happens at a later stage (typechecking or run time). One way to do it right is to write a `subst_span : Command -> (Span -> Span) -> Command` function that substitutes the spans. However, since this is the only case where we need to give spans right now manually, I just build the AST directly. In the future we can implement this subst_span function if needed.